open-sphere-base/mantle/src/main/java/io/opensphere/mantle/transformer/impl/worker/AbstractRenderPropertyUpdator.java

```diff
@@ -46,21 +42,24 @@ public DataElementTransformerWorkerDataProvider getProvider()
         return myProvider;
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public void run()
     {
-        RenderPropertyPool rpp = DefaultRenderPropertyPool.createPool(myProvider.getDataType(), myProvider.getGeometrySet(),
-                myProvider.getHiddenGeometrySet());
-        Set<RenderProperties> rpSet = rpp.values();
-        for (RenderProperties rp : rpSet)
+        // Get a set of unique render property objects
+        Map<RenderProperties, Object> renderPropertyMap = new IdentityHashMap<>();

```

Comment:
What is the point of using the identity hash map here? It appears that you're using it as a Set instead of a Map, so why not just use a Set to begin with?

Reply:
Because I need the identity aspect.  Before it was treating all render properties as the same if their equals/hashCodes matched, so it wasn't updating all of them, thus causing the issue.